### PR TITLE
Improve error message and exception type during `MultiprocessParallelUpdater` initialization

### PR DESCRIPTION
**Pull request by kmaehashi**
Added test for #4717.
Also changed from `ValueError` to `RuntimeError` as it is more appropraite.

### CODE REVIEW

**Comment by kmaehashi** [comment]
@mitmul  Do you have time to take a look on this (as it's related to #4717)? We discussed this in weekly meeting today; other team member can also do the review if you don't have time.

**Comment by mitmul** [comment]
@kmaehashi Sorry for delay. Could you check the flake8 errors in the travis tests?

File: tests/chainer_tests/training_tests/updaters_tests/snippets/cuda_init.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import numpy
+
+import chainer
+from chainer.backends import cuda
+from chainer import testing
```
Comment (okuta) [inline tests/chainer_tests/training_tests/updaters_tests/snippets/cuda_init.py]: ./tests/chainer_tests/training_tests/updaters_tests/snippets/cuda_init.py:5:1: F401 'testing' imported but unused

File: tests/chainer_tests/training_tests/updaters_tests/snippets/cuda_init.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+             chainer.datasets.split_dataset_n_random(
+                 dataset, len(devices))]
+    optimizer = chainer.optimizers.SGD(lr=1.0)
+    optimizer.setup(model)
+
+    # Initialize CUDA context.
+    cuda.cupy.cuda.runtime.runtimeGetVersion()
+
+    try:
+        updater = mpu.MultiprocessParallelUpdater(
```
Comment (okuta) [inline tests/chainer_tests/training_tests/updaters_tests/snippets/cuda_init.py]: ./tests/chainer_tests/training_tests/updaters_tests/snippets/cuda_init.py:26:9: F841 local variable 'updater' is assigned to but never used

**Comment by kmaehashi** [comment]
Thanks, fixed.

**Pull request merged**